src/app/[lng]/swap/sitemap.ts

```diff
@@ -0,0 +1,67 @@
+import type { MetadataRoute } from 'next';
+import { getChainsQuery } from '@/hooks/useChains';
+import type { Token } from '@lifi/sdk';
+import { getChainById } from '@/utils/tokenAndChain';
+import coins from '@/utils/coins';
+import { getSiteUrl } from '@/const/urls';
+
+//Optimized function to generate ordered bridge pairs (tokens from different chains)
+const generateBridgeOrderedPairs = (tokens: Token[]) => {
+  const orderedPairs: Array<[Token, Token]> = [];
+
+  // Loop through each token and compare with every other token
+  for (const token1 of tokens) {
+    for (const token2 of tokens) {
+      // Ensure tokens are from different chains (bridge rule)
+      if (token1.chainId !== token2.chainId) {
+        orderedPairs.push([token1, token2]); // Token1 -> Token2
+      }
+    }
+  }
+
+  return orderedPairs;
+};
+
+const sitemapLinksLimit = 50000;
+
+export async function generateSitemaps() {
+  const { chains } = await getChainsQuery();
+  const availableChainsId = chains.map((c) => c.id);
+
+  const ordered = generateBridgeOrderedPairs(
+    coins.filter((c) => availableChainsId.includes(c.chainId)) as Token[],
+  );
+
+  const numberOfChunks = Math.ceil(ordered.length / sitemapLinksLimit);
+
+  // Fetch the total number of products and calculate the number of sitemaps needed
+  return Array.from({ length: numberOfChunks }).map((_, index) => ({
+    id: index,
+  }));
+}
+
+export default async function sitemap({
+  id,
+}: {
+  id: number;
+}): Promise<MetadataRoute.Sitemap> {
+  const { chains } = await getChainsQuery();
+  const availableChainsId = chains.map((c) => c.id);
+
+  const ordered = generateBridgeOrderedPairs(
+    coins.filter((c) => availableChainsId.includes(c.chainId)) as Token[],
+  );
+
+  // Split the ordered array into chunks of 50,000
+  const start = id * sitemapLinksLimit;
+  const end = start + sitemapLinksLimit;
+  const orderedChunks = ordered.slice(start, end);
+
+  const routes = orderedChunks.map(([a, b]) => ({
+    url: `${getSiteUrl()}/bridge/${`${getChainById(chains, a.chainId)?.name}-${a.symbol}-to-${getChainById(chains, b.chainId)?.name}-${b.symbol}`.toLowerCase()}`,
```

Comment:
not good, seems to have bee copied without adaptation from the bridge

Reply:
It´s now removed